src/ship.js

```diff
@@ -0,0 +1,45 @@
+class Ship {
+  constructor(itinerary) {
+    this.itinerary = itinerary;
+    this.currentPort = itinerary.ports[0];
+    this.previousPort = null;
+    this.currentPort.addShip(this);
+  }
+
+  setSail() {
+    this.previousPort = this.currentPort;
+    this.currentPort = '';
+    this.previousPort.removeShip(this);
+  }
+
+  dock() {
+    const itinerary = this.itinerary;
+    const portIndex = itinerary.ports.indexOf(this.previousPort);
+    this.currentPort = itinerary.ports[portIndex + 1];
+    this.currentPort.addShip(this);
+  }
+}
+
+module.exports = Ship;
+
+// function Ship(itinerary) {
```

Comment:
It's good practice to delete commented out code in production software. I believe in this instance it's OK so you can keep both approaches as a reference.